### DISCUSSION TITLE
PageLoad request context for SSR

### DISCUSSION THREAD

**Discussion by austins** [sveltejs/kit#6899] [General]
Is there a way to get the `request` object (and/or `clientAddress`, `cookies`, `locals`, `platform`) like the `PageServerLoad` in  `PageLoad` (`+page.ts`) when it is server-side rendering on the initial request?

I'm working on a "hybrid" solution for [tRPC](https://trpc.io) where the initial request calls the procedures directly server-side and then any client-side navigated requests use the fetch API. The server-side calls would require the context (request, cookies, etc) of the server-side request. `+page.server.ts` is render blocking, hence the idea for this solution to improve the UX for subsequent requests that are client-side rendered.

**Comment by dummdidumm** [comment]
It's not possible to access them, `+page.js` is shared, so you can only access what exists in both.
I'm not sure what you mean by "render blocking"?
